### PR TITLE
Fixed bugs in JetMoE models.

### PR DESCRIPTION
**Pull request by Phoenix-Shen**
# What does this PR do?

Fixes issue #31791.
1. Fixed bugs in JetMoE casual language model and sequence classification model.
2. The original code will report errors when we want to output the `aux_loss`, like described in the issue.
3. I have modified `JetMoeForCausalLM` and `JetMoeForSequenceClassification`, now they work properly and output `aux_loss`, 
4. Meanwhile I have followed `MoeCausalLMOutputWithPast` class to write a new class named `MoESequenceClassifierOutputWithPast` to store the results of `JetMoeForSequenceClassification` model.
5. The code have passed the test of `tests/models/jetmoe/test_modeling_jetmoe.py`.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@ArthurZucker

### CODE REVIEW

**Comment by Phoenix-Shen** [comment]
`ci/circleci: tests_torch_and_tf` error doesn't seem to be a problem with the code I submitted.

**Comment by yikangshen** [comment]
The `check_code_quality` error can be fixed with this command:
`ruff format src/transformers/models/jetmoe/modeling_jetmoe.py`.
The `tests_torch_and_tf` will usually be automatically fixed after a resubmission.

**Comment by Phoenix-Shen** [comment]
> The `check_code_quality` error can be fixed with this command: `ruff format src/transformers/models/jetmoe/modeling_jetmoe.py`. The `tests_torch_and_tf` will usually be automatically fixed after a resubmission.

Thanks a lot, I have already formatted the files under your guidance.

**Review by ArthurZucker — COMMENTED** [review]
Thanks for updating! This LGTM, let's add a test the tests the loss computation with output router logits to make sure this is properly tested!
